### PR TITLE
Checkbox test - toggle off/on

### LINKED ISSUE #491: Checkbox Component Test
/components/Checkbox

### PR DESCRIPTION
**Pull request by SafeDuck**
image

### CODE REVIEW

File: cypress/component/Checkbox.cy.jsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import Checkbox from "@/components/Checkbox";
+import { mount } from "cypress-react-unit-test";
+
+describe("Checkbox Component Test", () => {
```
Comment (shahdivyank) [inline cypress/component/Checkbox.cy.jsx]:
```suggestion
describe("Checkbox", () => {
```

File: cypress/component/Checkbox.cy.jsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  it("toggle off", () => {
```
Comment (shahdivyank) [inline cypress/component/Checkbox.cy.jsx]:
```suggestion
  it("Toggle OFF", () => {
```

File: cypress/component/Checkbox.cy.jsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    cy.get('[data-cy="checkbox"]').should("exist");
+    cy.get('[data-cy="checkmark"]').should("exist");
+    cy.get('[data-cy="checkmark"]').should("have.class", "text-hackathon-gray");
+    cy.get('[data-cy="checkbox"]').should("have.class", "bg-hackathon-gray");
+  });
+
+  it("toggle on", () => {
+    let toggle = false;
+
+    const onClick = () => {
```
Comment (shahdivyank) [inline cypress/component/Checkbox.cy.jsx]: replicate this via hooks, take a look at the textarea tests

File: cypress/component/Checkbox.cy.jsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      const onClick = () => {
+        setToggle(!toggle);
+      };
+
+      return <Checkbox toggle={toggle} onClick={onClick} />;
+    };
+
+    cy.mount(<Parent />);
+
+    cy.get('[data-cy="checkbox-blue"]').click({ multiple: true });
```
Comment (shahdivyank) [inline cypress/component/Checkbox.cy.jsx]: remove the multiple true, it should only trigger one

File: src/components/Checkbox.jsx [hunk 5]
```diff
@@ -10,13 +10,15 @@ const Checkbox = ({ toggle, onClick = () => {}, text = "", color }) => {
       data-cy="checkbox"
     >
       <div
+        data-cy="checkbox-blue"
```
Comment (shahdivyank) [inline src/components/Checkbox.jsx]: why is it blue?

Comment (SafeDuck) [reply src/components/Checkbox.jsx]: I thought I changed that, i'll change it to "bg" which is what menthy suggested

**Review by shahdivyank — APPROVED** [review]

**Review by menthy-wu — APPROVED** [review]

**Pull request merged**